### ISSUE TITLE
Feature Request: Support SaveStates

### ISSUE THREAD

**Issue by sromeroi** [MiSTer-devel/MegaDrive_MiSTer#36]
It would be very helpful to have support to save/load savestates (like it's possible to do in NES' MiSTer core).

People tends now to have less time than 20 years ago to finish games and being able to save the current game state and come back to resume it some days later would be very a great feature :).

**Comment by birdybro** [comment]
Just to help manage yours and everyone else's expectations, savestates are very difficult to add to cores that weren't written with implementing them in mind and only one developer has ever added savestates to any cores (Robert Peip), it's not very simple to accomplish in FPGA.

**Comment by cheskote** [comment]
+1 for this feature request

**Comment by mambrino** [comment]
+1. It shouldn't be easy, but if there is a chance to do it, it would be very interesting.

**Comment by sromeroi** [comment]
> Just to help manage yours and everyone else's expectations, savestates are very difficult to add to cores that weren't written with implementing them in mind and only one developer has ever added savestates to any cores (Robert Peip), it's not very simple to accomplish in FPGA.

I completely understand it's a tough feature to add if the core has not been designed from the beginning with the savestates in mind.

But please understand that, from the user perspective, it is a very helpful feature and in several situations might be the reason for not to be able to transition from emulators to FPGAs. In my case, I rarely find more than 20-30 minutes at home to play some retro games, and due to this, I tend to use retropie-crt to be able to save and resume the game days later... It would be fantastic to be able to do it in MiSTer instead.

**Comment by SuperDannyG** [comment]
Sorry to revive this, but I've seen that SNES has save states added to its core and handles some of the enhancement chips as well.

As this was done to a core that wasn't designed for save states, could something like this be added to the Mega Drive core?

As the previous post mentioned, time is so limited these days that save states are a must.

**Comment by birdybro** [comment]
Of course, any core could have savestates added. It's just a lot harder if it wasn't made for it.
